src/compiler/core.ts

```diff
@@ -1134,10 +1134,33 @@ namespace ts {
      */
     export const supportedTypeScriptExtensions = [".ts", ".tsx", ".d.ts"];
     export const supportedJavascriptExtensions = [".js", ".jsx"];
-    const allSupportedExtensions  = supportedTypeScriptExtensions.concat(supportedJavascriptExtensions);
+
+    export function getExtensionsForGoal(goal: ResolutionGoal): string[] {

```

Comment:
Merged! `ExtensionPriority` overlays nicely onto what I used to call `ResolutionGoal` (now since renamed).